packaging/userspace/strip.py

```diff
@@ -0,0 +1,246 @@
+#!/usr/bin/env python3
+
+from pprint import pprint
+import inspect
+
+import shutil, sys, os, stat
+import logging
+import re
+import errno
+from os import path
+import argparse
+
+import script_common
+
+lief_imported = False
+lief_ver_good = False
+try:
+	import lief
+	lief_imported = True
+	lief_ver_good = script_common.check_lief_version()
+except ImportError:
+	pass
+
+re_unneeded_lib = re.compile("^\t(.+)$", re.MULTILINE)
+
+def strip(lib_path, env=None, strip_path=None, strip_debug=False, output_path=None, **kwargs):
+	try:
+		from irods import execute as ilib
+	except ImportError:
+		from irods import lib as ilib
+	l = logging.getLogger(__name__)
+
+	if output_path is not None and path.isdir(output_path):
+		output_path = path.join(output_path, path.basename(lib_path))
+		# if output_path is still a directory, panic
+		if path.isdir(output_path):
+			raise OSError(errno.EISDIR, "destination path is a directory.", output_path)
+
+	inplace = output_path is None or path.abspath(lib_path) == path.abspath(output_path)
+
+	if output_path is None:
+		output_path = lib_path
+
+	l.info(lib_path + ': Stripping unneeded symbols')
+
+	_env = os.environ.copy()
+	if env is not None:
+		for k, v in env.items():
+			_env[k] = v
+	env = _env
+
+	if strip_path is None:
+		strip_path = shutil.which('strip', path=env['PATH'])
+		if strip_path is None:
+			raise OSError(errno.ENOENT, "Could not find strip tool")
+
+	strip_args = [strip_path, '--strip-unneeded']
+	if strip_debug:
+		strip_args.append('--strip-debug')
+	strip_args.append(lib_path)
+	if not inplace:
+		strip_args.extend(['-o', output_path])
+
+	ilib.execute_command(strip_args, env=env)
+
+	os.chmod(output_path, stat.S_IRUSR | stat.S_IWUSR | stat.S_IXUSR |
+	                      stat.S_IRGRP |                stat.S_IXGRP |
+	                      stat.S_IROTH |                stat.S_IXOTH)
+
+	return output_path
+
+lib_size_warn_threshold = (36 * 1024 * 1024)
+if lief_ver_good and script_common.lief_ver < script_common.lief_min_rec_ver:
+	lib_size_warn_threshold = (24 * 1024 * 1024)
```

Comment:
are these known thresholds?   fine if so, but if these are our numbers, consider explaining and/or making them knobs/inputs...